### DISCUSSION TITLE
Could someone explain why aMule, gnutella and soulseek are still maintained but shareaza isn't?

### DISCUSSION THREAD

**Discussion by ghost** [nicotine-plus/nicotine-plus#2114] [Q&A] [answered]
I just can't understand how [aMule](https://github.com/amule-project/amule), [gnutella](https://github.com/gtk-gnutella/gtk-gnutella) and [soulseek](https://github.com/nicotine-plus/nicotine-plus) are still maintained but [shareaza](https://github.com/GetEnvy/Envy) isn't...
I mean why have three programs when we can have one and select which networks to use?
soulseek should have been added to shareaza too instead of doing another frontend with nicotine+

**Comment by mathiascode** [comment, marked as answer]
Different networks with different features and requirements, and clients with different goals. Shareaza does file transfers with IRC chat, while Soulseek has its own social aspect with chatrooms, private chat, user info, interests etc, which would be difficult to integrate into a general purpose file transfer program for multiple networks. Not to mention that the Soulseek protocol is vastly different from the other ones.

Open source programs come and go, depending on who has the time, interest and skill to maintain them. It's possible that someone will pick up Shareaza some day. Nicotine+ was barely maintained about two years ago, and was unmaintained for about 5 years prior to 2016, but I started contributing to it for personal reasons.

**Comment by slook** [comment]
> I just can't understand how [aMule](https://github.com/amule-project/amule), [gnutella](https://github.com/gtk-gnutella/gtk-gnutella) and [soulseek](https://github.com/nicotine-plus/nicotine-plus) are still maintained but [shareaza](https://github.com/GetEnvy/Envy) isn't...

If I were to hazard a guess, the developer of that project probably didn't have the resources to maintain all of the complex network protocols that needed to be supported, and perhaps didn't understand that maintaining support for so many different network protocols in one project is fundamentally impractical.

Nicotine was originally developed on and for Linux, which is based on the longest lasting operating system ever created "Unix", which has been around since the Epoch (which was on January 1st 1970 at 00:00)...

See link: [Basics of the Unix Philosophy](https://homepage.cs.uri.edu/~thenry/resources/unix_art/ch01s06.html) ...

_... Rule number 1:
"(i) Make each program do one thing well. To do a new job, build afresh rather than complicate old programs by adding new features."_
